static/app/views/performance/newTraceDetails/traceContextPanel.tsx

```diff
@@ -20,26 +20,19 @@ import type {TraceTreeNode} from 'sentry/views/performance/newTraceDetails/trace
 import {TraceViewLogsSection} from 'sentry/views/performance/newTraceDetails/traceOurlogs';
 
 type Props = {
+  logs: OurLogsResponseItem[] | undefined;
   onScrollToNode: (node: TraceTreeNode<TraceTree.NodeValue>) => void;
   rootEvent: UseApiQueryResult<EventTransaction, RequestError>;
   tree: TraceTree;
 };
 
-export function TraceContextPanel({tree, rootEvent, onScrollToNode}: Props) {
-  const renderTags = useCallback(() => {
-    if (!rootEvent.data) {
-      return null;
-    }
-
-    return (
-      <EventTagsTree
-        event={rootEvent.data}
-        meta={rootEvent.data._meta}
-        projectSlug={rootEvent.data.projectSlug ?? ''}
-        tags={rootEvent.data.tags ?? []}
-      />
-    );
-  }, [rootEvent.data]);
+export function TraceContextPanel({tree, rootEvent, onScrollToNode, logs}: Props) {
+  const hasProfiles = tree.type === 'trace' && tree.profiled_events.size > 0;
+  const hasLogs = logs && logs?.length > 0;
+  const hasTags =
+    rootEvent.data &&
+    rootEvent.data.tags.length > 0 &&
+    !(tree.type === 'empty' && hasLogs); // We don't show tags for only logs trace views
```

Comment:
nit: might be useful to break this out into a shareable function. Looks like the same logic is being used somewhere else in this pr too

Reply:
Fair point @edwardgou-sentry , I added a `useTraceContextSections` hook and reused it